projects/material-addons/src/lib/layout/action-bar/action-bar-layout.component.ts

```diff
@@ -0,0 +1,27 @@
+import {Component, Input} from '@angular/core';
+import {Router} from '@angular/router';
+import {Location} from '@angular/common';
+
+@Component({
+  selector: 'mad-actionbar-layout',
+  templateUrl: './action-bar-layout.component.html',
+  styleUrls: ['./action-bar-layout.component.scss']
+})
+export class ActionBarLayoutComponent {
+
+  @Input() title: string = '';
+  @Input() hasBackButton = true;
+
+
+  constructor(private router: Router, private location: Location) {
+  }
+
+
+  public goToPreviousPage(): void {
+    this.router.navigate([''], {skipLocationChange: true}).then(canRedirect => {
+      if (canRedirect) {
+        this.location.back();
+      }
+    });
```

Comment:
ich bin nicht sicher ob gute idee ist in library navigation machen

Reply:
aber ist ok für mich

Reply:
Denke, in diesem Fall, wo es nur um das "generische Back" geht, ist das vertretbar. Sonst hast du natürlich Recht!